manifests/package/install.pp

```diff
@@ -7,21 +7,17 @@
 #
 # * Justin Lambert <mailto:jlambert@letsevenup.com>
 #
-class artifactory::install {
+class artifactory::package::install inherits artifactory::params {
 
   if $caller_module_name != $module_name {
     fail("Use of private class ${name} by ${caller_module_name}")
   }
 
-  File {
-    require => Package['artifactory'],
```

Comment:
Yeah, not my favorite design pattern.  There isn't many File resources in this class - how about just throwing it on the individual resources?